lib/usecases/run/EndRunUseCase.js

```diff
@@ -50,8 +61,32 @@ class EndRunUseCase {
             const runEntity = await RunRepository.findOne(queryBuilder);
 
             if (runEntity) {
-                runEntity.timeO2End = timeO2End;
-                runEntity.timeTrgEnd = timeTrgEnd;
+                if (runQuality) {
+                    runEntity.runQuality = runQuality;
+                }
+                if (timeO2End && timeO2End > 0) {
+                    runEntity.timeO2End = timeO2End;
+                }
+                if (timeO2Start && timeO2Start > 0) {
+                    runEntity.timeO2Start = timeO2Start;
+                }
+                if (timeTrgEnd && timeTrgEnd > 0) {
+                    runEntity.timeTrgEnd = timeTrgEnd;
+                }
+                if (timeTrgStart && timeTrgStart > 0) {
+                    runEntity.timeTrgStart = timeTrgStart;
+                }
```

Comment:
I do not see the purpose of the `> 0` check. Is it because of the `-1`? Then it is even less obvious, maybe we should at least put a constant and test it exactly